p22/src/tictac.rs

```diff
@@ -0,0 +1,268 @@
+#[derive(Debug, Eq, PartialEq)]
+pub enum Error {
+    MoveOutBoard,
+    PositionTaken,
+}
+
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum StateBoard {
+    X,
+    O,
+    F,
+}
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum StateGame {
+    WinX,
+    WinO,
+    WinBoth,
+    GameOn,
+}
+#[derive(Debug, Clone, Copy)]
+pub struct Player {
+    pub mark: StateBoard,
+}
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct TicTacField {
+    pub board: [StateBoard; 9],
```

Comment:
To simplify code a bit you could use nested arrays, i.e. `[[StateBoard; 3]; 3]`.